specifyweb/frontend/js_src/lib/components/QueryBuilder/Edit.tsx

```diff
@@ -87,7 +86,7 @@ function EditQueryDialog({
       onAdd={undefined}
       onClose={handleClose}
       onDeleted={handleClose}
-      onSaved={(): void => navigate(`/specify/query/${queryResource.id}/`)}
+      onSaved={(): void => window.location.reload()}
     >
```

Comment:
Might be good to use `globalThis` here since other places use that for reloading
```
globalThis.location.reload()
```